jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/listener/JMSConnectionRetryHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.transport.jms.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.transport.jms.exception.JMSConnectorException;
+
+/**
+ * This class tries to connect to JMS provider until the maximum re-try count meets.
+ */
+class JMSConnectionRetryHandler {
+    private JMSServerConnector jmsServerConnector;
+    private static final Log logger = LogFactory.getLog(JMSConnectionRetryHandler.class.getName());
+    private long retryInterval;
+    private int retryCount = 0;
+    private int maxRetryCount;
+
+    /**
+     * Creates a jms connection retry handler.
+     * @param jmsServerConnector JMS Server Connector
+     * @param retryInterval Retry interval between
+     * @param maxRetryCount Maximum retries
+     */
+    JMSConnectionRetryHandler(JMSServerConnector jmsServerConnector, long retryInterval, int maxRetryCount) {
+        this.jmsServerConnector = jmsServerConnector;
+        this.retryInterval = retryInterval;
+        this.maxRetryCount = maxRetryCount;
+    }
+
+    /**
+     * To start the retrying to connect to JMS provider.
+     * @throws JMSConnectorException JMS Connector Exception
+     */
+    void start() throws JMSConnectorException {
+        logger.error("Re-connection will be attempted after " + retryInterval + " milli-seconds.");
+        try {
+            Thread.sleep(retryInterval);
+        } catch (InterruptedException e) {
+            //Ignore the exception
```

Comment:
Shouldn't we restore the interrupted status? 
https://www.ibm.com/developerworks/library/j-jtp05236/